### DISCUSSION TITLE
Internet Explorer has been dropped from rendered compat tables on MDN

### DISCUSSION THREAD

**Discussion by queengooborg** [mdn/mdn-community#202] [Browser compatibility data] [answered]
Hey all!

Earlier this week, a [pull request](https://github.com/mdn/yari/pull/6602) was merged to hide Internet Explorer from the compatibility tables on MDN Web Docs.  This decision was made in response to the official consumer End of Life of Internet Explorer[^1][^2] on **June 15, 2022**, as well as the fact that worldwide market share for IE has dropped to **0.28%** (as of July 2022)[^3] and has a maximum region-specific usage of **1.66%** in China[^4], far below any of the other browsers we record BCD for.  Additionally, in China, Chrome-based browsers take the lead with a combined 76.22% market share[^5].

With the official EOL, the declining usage statistics, and the countless security vulnerabilities that will never be fixed in the browser, we decided that it was time for us to **no longer display IE** compatibility data as to **discourage web developers** from writing IE-compatible websites.  We highly encourage web developers to port their IE-only websites to modern browsers like Chrome, Firefox and Safari.

Note that the raw data is still available for the time being in the [browser-compat-data](https://github.com/mdn/browser-compat-data) repository[^6], and is currently still accessible through the Node.js package and partner sites like https://caniuse.com/ for those who absolutely need the compatibility data.  However, the data should be considered legacy data as it will **not** be maintained and will eventually be removed entirely.

[^1]: https://death-to-ie11.com/
[^2]: https://docs.microsoft.com/en-us/lifecycle/announcements/internet-explorer-11-end-of-support#:~:text=Internet%20Explorer%20(IE)%2011%20is,certain%20versions%20of%20Windows%2010*.
[^3]: https://gs.statcounter.com/browser-market-share
[^4]: https://canistop.net/b/ie/11
[^5]: https://gs.statcounter.com/browser-market-share/all/china
[^6]: https://github.com/mdn/browser-compat-data/issues/15333

**Comment by Josh-Cena** [comment]
As a developer from China, I have to say we (general-public-facing site developers) will likely maintain compatibility with IE for a few years—there's no clear sign of getting off it yet (although MDN also made similar decisions to get rid of other browsers that are only popular in certain regions, so not sure how much influence that is). "Compatible with IE" remains to be a recurrent theme among Chinese developers I've engaged with.

I don't have much to share, though. It's been long since I maintained a Chinese-audience website.

**Reply by queengooborg** [reply]
We’re actually planning on adding back the browsers with a more Eastern focus, like UC and QQ.  The reason we had removed those before wasn’t actually to do with global market share, but rather a lack of maintainers updating its data.  Now that we’ve implemented build-time mirroring, where we can copy the support data from another browser, we’re planning to bring them back soon!

**Reply by teoli2003** [reply]
So the plan is to add them back to mdn/browser-compat-data. Is there also a plan to bring them back to the table displayed on MDN?

**Reply by queengooborg** [reply]
We’re currently discussing a way to group browsers with matching engines together and only show the differences between them!

**Comment by mass10** [comment]
Hi,

We think, no more new applications for Internet Explorer might be developed anymore.

But I known many developers maintaining applications runnning for Internet Explorer today.

I do hope https://github.com/mdn/yari/pull/6602 will be reverted.

**Reply by juls0730** [reply]
It really amazes me that people still support a browser that's latest release (IE11) is almost 9 years old and has major security issues. It's just amazing to me that developers aren't forcing users to get off of IE11.

**Reply by Josh-Cena** [reply]
Unfortunately reality doesn't work that way. Developers are paid by their employers, and their employers profit from their users. If you upset your users you are not going to get money from them. There's no point in fighting against money😉

It's the same for "don't break the web". Can browsers _force_ developers to upgrade their site to amend for some breaking change, even when it makes everyone better off? They can't, because it's not a monopoly and your users simply leave for other alternatives.

**Reply by mass10** [reply]
I agree that I wat no more IEs remain. But, many kind of environments exist, actually.

thank you for your comment.

**Comment by zypA13510** [comment]
I don't think any developers in their right mind would **WANT** to support IE. You'd think by now most people remotely involved in web development should know IE is just outdated, buggy, and above all, insecure. We don't need MDN to remind us this fact. But we just **NEED** to support IE and it's not our decision to make.

If MDN is going to hide that information from us, it simply means we now have to go to another source to check IE compatibility. In another word, you are making it less useful to part of your audience, without making it - in any way I can see - more useful to anyone else. So why? Why you do this to us?

P.S. Also, 1 percent is still a very significant number when you are talking about billions of browsers.

**Reply by juls0730** [reply]
I’m not entirely sure if I would agree, as you said IE is “outdated, buggy, and above all, insecure” and your right, so why do you “**NEED** to support IE”? The only reason IE is still used by about 50 million people for some reason is because developers are not forcing them off as any sane developer should. Supporting IE would be as crazy as developers still supporting the original ds, even Nintendo decided not to. IE reached EOL back in June, you should’ve started moving users off of IE even sooner than that as IE has been a lost cause for many years before that.

**Reply by Josh-Cena** [reply]
When was the last time developers single-handedly "forced" users to do something? The tech market to this day is still a buyer's market with considerable competition. If you (assuming you are employed to a socially impactful industry like insurance or hospital) decide to not support IE, you are putting _yourself_ at a disadvantage. Can't stress this more—developers are the stakeholders with the least voice in this situation. We have employers. We have users whom we literally feed on and try to make happy. We have product managers who know nothing about tech but the fact that our users are upset because they can't use the browsers they always use. It's just not our decision to make.

You may think the browser vendors are powerful enough. So why not try persuading Chrome/Firefox to drop support for {`Symbol.species`, `fn.caller`, `__proto__`}, because it's so {insecure, inelegant, ill-performing}? They cannot because it instantly means they lose users—even for something as seemingly dominant as Chrome.

To be clear, I'm happy as anyone else to get rid of every trace of the existence of IE. My own websites only support evergreen browsers because I know who my audience is. But if you've worked in the tech industry for a while, you will feel a sense of powerlessness of yourself. You are only a worker. You follow the roadmap set by your manager. You work to fulfill your KPIs every month. You implement whatever ridiculous features your employer wants. Heck, you can't even decide whether to use single quotes or not! MDN is a developer tool. We are not there to make judgement calls when things aren't in our hands.

**Reply by juls0730** [reply]
As developers, hasn’t it always been our job to protect the users? If there is a security vulnerability in your application it’s your job to fix it, no? I (a developer of 4 years who has never supported IE) do not see why we should support our users who use a browser with major vulnerabilities. I can 100% see if your a developer in a certain geography that IE is used as you (or used to) do. But I do not think we should be supporting users who use these browsers.

**Reply by zypA13510** [reply]
@juls0730

> As developers, hasn’t it always been our job to protect the users?

If you ever worked for a large company, you know that is **not** your job. There are other people for that. Your job as a developer is to simply **implement** what other people have in mind, you are not there to make decisions. You either do your job, or lose it.

> If there is a security vulnerability in your application it’s your job to fix it, no?

Yes, but only if the security people say that this is a vulnerability. If they don't, your product manager is going to push you to fix these incompatibilities.

Like I said, it's simply not our decision to make.

**Comment by Rumyra** [comment, marked as answer]
Thanks for your feedback everyone - just a note to say we're planning on doing some more research in this area before making anymore updates to the compat tables on MDN - this should happen in the coming months and we'll share insights 👍

**Comment by aleen42** [comment]
OMG, I don't think it is a good decision. There are still a lot of developers who maintain programs under IEs depending on the MDN document. 😨

**Reply by teoli2003** [reply]
I'm curious here:
- What information do you use from MDN? Info about the availability of a feature you want to use in IE? Or information about IE-only features?
- Are you using a transpiler? If not, why?
- Are you using polyfills? If so, where do you look for them
- Other use cases?
- Can you use graceful degradation for IE users or do you have to provide the same functionality to every browser you support?
- Is the IE site available on the Web, or only in an intranet?
- Is there already a plan to move away from IE or is it expected to be like this for years to come?

- Do you know of alternative places (sites, books) to find the info you need?

**Reply by aleen42** [reply]
> What information do you use from MDN? Info about the availability of a feature you want to use in IE? Or information about IE-only features?

Of course, the compatibility info to check whether we can use under IE, like a CSS feature which can not be patched. For example, can we 
 use `cursor: not-allowed` under IE5.5?

> Are you using a transpiler? If not, why?

> Are you using polyfills? If so, where do you look for them?

I knew most shims like [Babel](https://github.com/babel/babel) (mainly for ES syntax transformation), [ES3-compatible-webpack-plugin](https://github.com/aleen42/ES3-compatible-webpack-plugin) (mainly compatible for ES3 syntax), [core-js](https://github.com/zloirock/core-js) (mainly for JS standard APIs), [core-web](https://github.com/aleen42/core-web) (mainly for Web APIs, which I maintain right now), [autoprefixer](https://github.com/postcss/autoprefixer) (mainly for add vendor prefixers for CSS rules), [css-split-webpack-plugin)](https://github.com/bootstarted/css-split-webpack-plugin) (mainly for spliting CSS rules to avoid the number limitation under IE), etc.

All those libraries' maintainers may always need to know the compatibility of IE, and MDN has made it less helpful, why?

> Other use cases?

I [contributed](https://github.com/mdn/browser-compat-data/pulls?q=is%3Apr+author%3Aaleen42+is%3Aclosed) MDN to describe some edge cases for developers to avoid.

> Can you use graceful degradation for IE users or do you have to provide the same functionality to every browser you support?

In my company (which has provided Email services for different companies or organizations), most of these users still have a requirement under IEs because of their old facilities. so we currently still need support for IE8+. Some basic functionalities like an old login page should even support below IE8.

> Is the IE site available on the Web, or only on an intranet?

Definitely on the web.

> Is there already a plan to move away from IE or is it expected to be like this for years to come?

There is no plan currently or even in these few years unless Microsoft has a mechanism to force users to upgrade old systems like Windows XP, or IE to Edge. I don't think it is possible.

> Do you know of alternative places (sites, books) to find the info you need?

Currently, I think MDN is the best choice before eliminating the information.

In my opinion, MDN should record all information on technicals of Web development, including JS, Web APIs, HTML, CSS, etc. I don't think it is a huge overhead for you to do so because IE has no featured updating.

**Reply by teoli2003** [reply]
Thank you for your time! This will help me in forging my personal opinion about all this.
